src/MultiDimFit.cc

```diff
@@ -135,7 +138,7 @@ void MultiDimFit::applyOptions(const boost::program_options::variables_map &vm)
     skipInitialFit_ = (vm.count("skipInitialFit") > 0);
     hasMaxDeltaNLLForProf_ = !vm["maxDeltaNLLForProf"].defaulted();
     loadedSnapshot_ = !vm["snapshotName"].defaulted();
-    savingSnapshot_ = (!loadedSnapshot_) && vm.count("saveWorkspace");
+    savingSnapshot_ = vm.count("saveWorkspace");
```

Comment:
make it possible to override an existing loaded snapshot. @nucleosynthesis - ok for you?